### PR TITLE
Fix batch for D10.1

### LINKED ISSUE #5494: Fix batch for Drupal 10.1
**Is your feature request related to a problem? Please describe.**
Core getting rid of using `sequences` table and batch now using auto-increment https://www.drupal.org/node/3220378

Needs to fix [this pace at least](https://github.com/drush-ops/drush/blob/2b890ab5c2b5d70cae7b17ebda63ad57a3b318ab/includes/batch.inc#L129) the same way like https://git.drupalcode.org/project/drupal/-/commit/779785e220253c1d9fee8c43f7c7ed343a64244e

**Describe the solution you'd like**
Use to check if method exists and fallback to database method for BC

**Describe alternatives you've considered**
no

**Additional context**
running updates is broken https://www.drupal.org/project/drupal/issues/3337513#comment-14978772

### PR DESCRIPTION
**Pull request by mondrake**
Fixes #5494

To test this, https://www.drupal.org/project/drupal/issues/3337513 should be merged in Drupal core, OR its MR applied.

### CODE REVIEW

File: includes/batch.inc [hunk 1]
```diff
@@ -126,7 +128,19 @@ function _drush_backend_batch_process($command = 'batch-process', $args = [], $o
 
     // Assign an arbitrary id: don't rely on a serial column in the 'batch'
     // table, since non-progressive batches skip database storage completely.
-    $batch['id'] = \Drupal::database()->nextId();
+    /** @var \Drupal\Core\Batch\BatchStorageInterface $batch_storage */
+    $batch_storage = \Drupal::service(BatchStorageInterface::class);
```
Comment (andypost) [inline includes/batch.inc]: Aliases been added in 10.1 so this will fail on earlier core, please use `batch.storage` as service name

Comment (andypost) [reply includes/batch.inc]: dgo.to/3049525

**Comment by weitzman** [comment]
PRs should go into 12.x branch first. That branch is Drupal 10 only. Thanks.

**Comment by mondrake** [comment]
Hopefully retargeted correctly.

File: docs/install.md [hunk 2]
```diff
@@ -17,6 +17,13 @@
 Drupal Compatibility
 -----------------
 <table>
+  <tr>
+    <td> Drush 12 </td>
```
Comment (andypost) [inline docs/install.md]: looks  rebase artefact

Comment (mondrake) [reply docs/install.md]: thanx, fixed

File: includes/batch.inc [hunk 3]
```diff
@@ -105,7 +107,19 @@ function _drush_backend_batch_process($command = 'batch-process', $args = [], $o
...
     // table, since non-progressive batches skip database storage completely.
-    $batch['id'] = \Drupal::database()->nextId();
+    /** @var \Drupal\Core\Batch\BatchStorageInterface $batch_storage */
+    $batch_storage = \Drupal::service('batch.storage');
+    if (method_exists($batch_storage, 'getId')) {
+      try {
+        $batch['id'] = $batch_storage->getId();
+      }
+      catch (IntegrityConstraintViolationException $e) {
+        $batch['id'] = \Drupal::database()->nextId();
```
Comment (weitzman) [inline includes/batch.inc:117]:
```suggestion
        // Remove calls to nextId() once 10.0 is no longer supported.
        $batch['id'] = \Drupal::database()->nextId();
```

**Comment by ddaffie** [comment]
@weitzman Thanks!

**Comment by andypost** [comment]
Is there a way to backport it to 11.x?

**Pull request merged**